internal/bridgeapi/client.go

```diff
@@ -0,0 +1,308 @@
+/*
+Copyright 2021 Crunchy Data Solutions, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package bridgeapi
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/go-logr/logr"
+)
+
+var (
+	routeClusters    string = "/clusters"
+	routeDefaultRole string = "/clusters/%s/roles/default"
+)
+
+// Leave open configuration for eventual-consistency scenario
+type Client struct {
+	APITarget *url.URL
+	Log       logr.Logger
+	client    *http.Client
+}
+
+// NewClient returns a client initialized with the package logger as the
+// default logger and an uninitialized APITarget for late binding
+func NewClient() *Client {
+	client := &Client{
+		Log: pkgLog,
+	}
+	return client
+}
+
+func (c *Client) precheck() error {
+	if c.APITarget == nil {
+		return ErrorAPIUnset
+	}
+
+	// Lazy initialization
+	if c.client == nil {
+		c.client = &http.Client{}
+	}
+
+	return nil
+}
+
+// helper to set up auth with current bearer token
+func setBearer(req *http.Request) {
+	req.Header.Set("Authorization", "Bearer "+primaryLogin.activeToken)
+}
+
+func (c *Client) CreateCluster(cr CreateRequest) error {
+	if err := c.precheck(); err != nil {
+		return err
+	}
+	// TODO: Identify personal team id if not provided in request
+
+	reqPayload := new(bytes.Buffer)
+	json.NewEncoder(reqPayload).Encode(cr)
+	req, err := http.NewRequest(http.MethodPost, c.APITarget.String()+routeClusters, reqPayload)
```

Comment:
I'm not sure I understand the question. In this code, APITarget is of `url.URL` type so its Stringer interface is invoked to use the string concatenation for the request URL.

Reply:
my question was it does it return the correct Base URL,  might be change into`APITarget.Host`  `https://golang.org/pkg/net/url/#example_URL` but if returning Base URL then fine no issue, thanks

Reply:
I see what you're asking now. In its present form, it is relying on APITarget being correctly set to the base URL for the API. I'm inclined to leave that as more of a documentation issue to make clear what value is expected there instead of parsing specific sub-fields of the URL, which in this case would be a minimum of scheme, host, and path.

Reply:
btw we can also simple string contain the URL-> https://api.crunchybridge.com

Reply:
I saw that change in your draft set - I'm enforcing URL to force the type instead of the library doing the work to verify the string it was given is a valid URL.

Reply:
okay, got it.